src/generator/pollers.ts

```diff
@@ -70,7 +134,59 @@ export async function generatePollers(session: Session<CodeModel>): Promise<stri
     let handleResponse = '';
     const schemaResponse = <SchemaResponse>poller.op.responses![0];
     let unmarshalResponse = 'nil';
-    if (isSchemaResponse(schemaResponse) && schemaResponse.schema.language.go!.responseType.name !== undefined) {
+    let pagerFields = '';
+    if (isPageableOperation(poller.op)) {
+      responseType = poller.op.language.go!.pageableType.name;
+      pagerFields = `
+      respHandler ${camelCase(poller.op.language.go!.pageableType.op.responses![0].schema.language.go!.name)}HandleResponse`;
+      pollUntilDoneResponse = `(${responseType}, error)`;
+      pollUntilDoneReturn = 'p.FinalResponse(ctx)';
+      // for operations that do return a model add a final response method that handles the final get URL scenario
+      finalResponseDeclaration = `FinalResponse(ctx context.Context) (${responseType}, error)`;
+      finalResponse = `FinalResponse(ctx context.Context) (${responseType}, error) {
```

Comment:
This should be refactored to avoid the duplication.

Reply:
Done